src/mumble/GlobalShortcut_xdp.cpp

```diff
@@ -0,0 +1,189 @@
+
+#include <QtGui/QWindow>
+
+Q_GLOBAL_STATIC_WITH_ARGS(OrgFreedesktopPortalGlobalShortcutsInterface, s_shortcutsInterface, (QLatin1String("org.freedesktop.portal.Desktop"),
+                                                                        QLatin1String("/org/freedesktop/portal/desktop"),
+                                                                        QDBusConnection::sessionBus()));
+
+bool GlobalShortcutXdp::isAvailable()
+{
+	return s_shortcutsInterface->isValid();
+}
+
+GlobalShortcutXdp::GlobalShortcutXdp() {
+	qDBusRegisterMetaType<XdpShortcuts>();
+	qDBusRegisterMetaType<QPair<QString, QVariantMap>>();
+
+	connect(s_shortcutsInterface, &OrgFreedesktopPortalGlobalShortcutsInterface::Activated, this, [this] (const QDBusObjectPath &session_handle, const QString &shortcut_id, qulonglong timestamp, const QVariantMap &options) {
+        Q_UNUSED(session_handle);
+        Q_UNUSED(timestamp);
+        Q_UNUSED(options);
+
+        foreach (GlobalShortcut *shortcut, qmShortcuts) {
+            if (shortcut_id == shortcut->objectName()) {
+                shortcut->triggered(true, shortcut_id);
+                shortcut->down(shortcut_id);
+            }
+        }
+    });
+    connect(s_shortcutsInterface, &OrgFreedesktopPortalGlobalShortcutsInterface::Deactivated, this, [this] (const QDBusObjectPath &session_handle, const QString &shortcut_id, qulonglong timestamp, const QVariantMap &options) {
+        Q_UNUSED(session_handle);
+        Q_UNUSED(timestamp);
+        Q_UNUSED(options);
+
+        foreach (GlobalShortcut *shortcut, qmShortcuts) {
+            if (shortcut_id == shortcut->objectName()) {
+                shortcut->triggered(false, shortcut_id);
+            }
+        }
+    });
+    connect(s_shortcutsInterface, &OrgFreedesktopPortalGlobalShortcutsInterface::ShortcutsChanged, this, [this] (const QDBusObjectPath &session_handle, const QList<QPair<QString,QVariantMap>> &shortcuts) {
+        Q_UNUSED(session_handle);
+        if (m_shortcuts != shortcuts) {
+            m_shortcuts = shortcuts;
+            Q_EMIT shortcutsChanged();
+        }
+    });
+
+    QTimer::singleShot(0, this, &GlobalShortcutXdp::createSession);
+}
+
+GlobalShortcutXdp::~GlobalShortcutXdp() {
+
+}
+
+void GlobalShortcutXdp::createSession()
+{
+    XdpShortcuts initialShortcuts;
+    initialShortcuts.reserve(qmShortcuts.count());
+    Global::get().s.qlShortcuts.clear();
+
+    int i = 0;
+    m_ids.resize(qmShortcuts.count());
+    foreach (GlobalShortcut *shortcut, qmShortcuts) {
+        initialShortcuts.append({shortcut->objectName(), {
+            { QStringLiteral("description"), shortcut->name }
+        }});
+
+        Shortcut ssss = { i, {uint(i)}, shortcut->qvDefault, false};
+        m_ids[i] = shortcut->objectName();
+        Global::get().s.qlShortcuts << ssss;
+        ++i;
+    }
+
+    QDBusArgument arg;
+    arg << initialShortcuts;
+    auto reply = s_shortcutsInterface->CreateSession({
```

Comment:
Please use the actual type instead of `auto` (see our coding guidelines for details)